File: packages/shared/src/types/evidence.ts
import type { EvidencePackId, WorkspaceId, SourceId, VaultNoteId } from "./ids.js";

export interface EvidenceExcerpt {
  readonly sourceId: SourceId;
  readonly chunkIndex: number;
  readonly text: string;
  readonly score: number;
  readonly locationHint: string | null;
}

export interface EvidencePack {
  readonly id: EvidencePackId;
  readonly workspaceId: WorkspaceId;
  readonly query: string;
  readonly excerpts: readonly EvidenceExcerpt[];
  readonly noteRefs: readonly VaultNoteId[];
  readonly model: string | null;
  readonly generatedAt: string;
}

File: services/api/src/atlas_api/models/evidence.py
"""Evidence domain models — mirrors packages/shared/src/types/evidence.ts."""

from pydantic import BaseModel, ConfigDict


class EvidenceExcerpt(BaseModel):
    model_config = ConfigDict(frozen=True)

    source_id: str
    chunk_index: int
    text: str
    score: float
    location_hint: str | None


class EvidencePack(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    query: str
    excerpts: tuple[EvidenceExcerpt, ...]
    note_refs: tuple[str, ...]
    model: str | None
    generated_at: str
